pkg/kubeapiserver/server/insecure_handler.go

```diff
@@ -30,14 +30,14 @@ import (
 func BuildInsecureHandlerChain(apiHandler http.Handler, c *server.Config) http.Handler {
 	handler := apiHandler
 	handler = genericapifilters.WithAudit(handler, c.AuditBackend, c.AuditPolicyChecker, c.LongRunningFunc)
-	handler = genericapifilters.WithAuthentication(handler, server.InsecureSuperuser{}, nil, nil)
-	handler = genericfilters.WithCORS(handler, c.CorsAllowedOriginList, nil, nil, nil, "true")
-	handler = genericfilters.WithTimeoutForNonLongRunningRequests(handler, c.LongRunningFunc, c.RequestTimeout)
 	if feature.DefaultFeatureGate.Enabled(features.RequestManagement) {
 		handler = genericfilters.WithRequestManagement(handler, c.LongRunningFunc, c.RequestManagement)
 	} else {
 		handler = genericfilters.WithMaxInFlightLimit(handler, c.MaxRequestsInFlight, c.MaxMutatingRequestsInFlight, c.LongRunningFunc)
 	}
+	handler = genericapifilters.WithAuthentication(handler, server.InsecureSuperuser{}, nil, nil)
```

Comment:
Very good!  It should have been this way from the start.  Perhaps it will solve our problem with lack of user info in the insecure case.